ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -133,11 +132,18 @@ class IndexSegment implements Iterable<IndexEntry> {
     this.factory = factory;
     this.metrics = metrics;
     this.time = time;
+    // When PersistentIndex tries to add one IndexEntry and it decides to create a new IndexSegment,
+    // it provides that new IndexEntry's valueSize and entrySize for the IndexSegment.
```

Comment:
is entrySize = keySize + valueSize ?

Reply:
Yes, the JIRA ticket I mentioned have some explanation of it.